internal/build/phase_config_provider.go

```diff
@@ -159,7 +184,16 @@ func WithLifecycleProxy(lifecycleExec *LifecycleExecution) PhaseConfigProviderOp
 
 func WithNetwork(networkMode string) PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
-		provider.hostConf.NetworkMode = container.NetworkMode(networkMode)
+		if networkMode != "" {
+			if provider.hostConf.NetworkMode != "" {
+				w := provider.errorWriter
+				if w == nil {
+					w = os.Stderr
+				}
+				fmt.Fprintf(w, "Overriding network from %q to %q.", provider.hostConf.NetworkMode, networkMode)
```

Comment:
This seems to be more of a debug message than an error message. This, together with the comment above, may point us in the direction of threading the logger directly into the provider.